### PR TITLE
Add trigger options to weapon events

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Add options to weapon events / weapon animations framework:
  - Leave Optics: leave optics view to bolt/rack (old default)
  - Trigger Release: release trigger to bolt/rack
  - Trigger Press: press trigger again to bolt/rack

TODO:
- [x] Make `CBA_triggerPressed` GVAR or public API -> GVAR
- [x] Set default option to Trigger Release
  - Friendlier to new users and used in more games
- [X] Support non-LMB triggers -> **later if requested** (if anyone even uses non-LMB trigger)
- [X] Add option for vanilla behaviour (Trigger Release default accomplishes the same) -> **later if requested**
- [X] Rename "Trigger Press" to "Trigger Re-Press" -> **nope**

### CODE REVIEW

File: addons/events/XEH_postInit.sqf [hunk 1]
```diff
@@ -66,3 +66,23 @@ if (isServer) then {
...
+    if (_key == 0) {
+        CBA_triggerPressed = true;
+    };
+}] call CBA_fnc_addDisplayHandler;
+
+["MouseButtonUp", {
+    params ["", "_key"];
+
+    // TODO Support non-LMB (?)
+    if (_key == 0) {
```
Comment (commy2) [inline addons/events/XEH_postInit.sqf]: It should be `_button` not `_key`, right? "mouse button"

File: addons/events/fnc_weaponEvents.sqf [hunk 2]
```diff
@@ -90,8 +93,18 @@ if (!_isEmpty || _onEmpty) then {
...
         if (count magazines _unit != _expectedMagazineCount) exitWith {true};
 
-        // while in gunner view, keep waiting
-        if (cameraView == "GUNNER" && _optic != "") exitWith {
+        // mode 0: while in gunner view, keep waiting
+        // mode 1: while holding trigger, keep waiting
+        // mode 2: while holding trigger and not pressing it, keep waiting
+        private _wait = (GVAR(weaponEventMode) == 0 && cameraView == "GUNNER" && _optic != "") ||
+                        {GVAR(weaponEventMode) == 1 && CBA_triggerPressed} ||
+                        {GVAR(weaponEventMode) == 2 && (_trigger || !CBA_triggerPressed)};
```
Comment (commy2) [inline addons/events/fnc_weaponEvents.sqf]: Optimize this to:
```sqf
        private _wait = call ([{
            cameraView == "GUNNER" && _optic != ""
        }, {
            CBA_triggerPressed
        }, {
            _trigger || !CBA_triggerPressed
        }] select GVAR(weaponEventMode));
```
They are currently quite a lot of logic operators.

**Comment by jonpas** [comment]
Done.

I decided to not support non-LMB triggers or add vanilla behaviour at this time, reasons above. Can be added later if requested.

Should I rename "Trigger Press" to "Trigger Re-Press" to make it clearer or is it clear enough?

**Comment by jonpas** [comment]
On hold for @commy2's settings update that will allow adding settings in preInit regardless of load order.

**Comment by commy2** [comment]
Note to self, pls ignore:

>cba_settings_fnc_init could push to some array if the settings preInit hasn't run yet. And that array is evaluted once settings preInit runs.

File: addons/events/initSettings.sqf [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    "LIST",
+    [LLSTRING(WeaponEventMode), LLSTRING(WeaponEventModeTooltip)],
+    LLSTRING(Category),
+    [[0, 1, 2], [
+        [LLSTRING(WeaponEventModeOptic), LLSTRING(WeaponEventModeOpticTooltip)], // Exit optic view
+        [LLSTRING(WeaponEventModeTriggerRelease), LLSTRING(WeaponEventModeTriggerReleaseTooltip)], // Stop holding trigger
+        [LLSTRING(WeaponEventModeTriggerPress), LLSTRING(WeaponEventModeTriggerPressTooltip)] // Click trigger again
+    ], 1],
+    2
+] call (uiNamespace getVariable "cba_settings_fnc_init");
```
Comment (commy2) [inline addons/events/initSettings.sqf]:
```suggestion
] call CBA_fnc_addSetting;
```
Once #1111 is merged.

**Pull request merged**